### PR TITLE
Call recording docs

### PR DESCRIPTION
**Pull request by jamesaimonetti**

### CODE REVIEW

File: applications/crossbar/priv/couchdb/schemas/callflows.record_call.json [hunk 1]
```diff
@@ -4,55 +4,70 @@
...
             "type": "boolean"
         },
         "record_sample_rate": {
-            "description": "Sampling rate of the recording, in Hz",
+            "description": "What sampling rate to use on the recording",
             "name": "Sample Rate",
             "required": false,
             "type": "integer"
         },
+        "recording_id": {
```
Comment (lazedo) [inline applications/crossbar/priv/couchdb/schemas/callflows.record_call.json]: this is not used. recording_id is always generated

File: applications/crossbar/priv/couchdb/schemas/callflows.record_call.json [hunk 2]
```diff
@@ -4,55 +4,70 @@
...
             "name": "Time Limit",
             "required": false,
             "type": "integer"
         },
         "url": {
-            "description": "What URL to use as a base for where to send the recording after it finishes",
+            "description": "The URL to use when sending the recording for storage",
             "name": "URL",
-            "required": false,
+            "required": true,
```
Comment (lazedo) [inline applications/crossbar/priv/couchdb/schemas/callflows.record_call.json]: this is not required, if provided will save to url, if not saves locally. i would actually like to remove this and have it set at the data manager level.

File: applications/crossbar/priv/couchdb/schemas/callflows.record_caller.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        "time_limit": {
+            "default": 3600,
+            "description": "Time limit, in seconds, for the recording",
+            "maximum": 3600,
+            "minimum": 5,
+            "name": "Time Limit",
+            "required": false,
+            "type": "integer"
+        },
+        "url": {
```
Comment (lazedo) [inline applications/crossbar/priv/couchdb/schemas/callflows.record_caller.json]: this is not requred

File: core/whistle_media/src/wh_media_recording.erl [hunk 4]
```diff
@@ -110,7 +110,7 @@
 -define(CONSUME_OPTIONS, []).
 -define(MAX_RECORDING_LIMIT, wh_media_util:max_recording_time_limit()).
 -define(CHECK_CHANNEL_STATUS_TIMEOUT, 5 * ?MILLISECONDS_IN_SECOND).
--define(RECORDING_ID_KEY, <<"Recording-ID">>).
+-define(RECORDING_ID_KEY, [<<"recording_id">>, <<"Recording-ID">>]).
```
Comment (lazedo) [inline core/whistle_media/src/wh_media_recording.erl]: `recording_id` shouldn't be used.

File: core/whistle_media/src/wh_media_recording.erl [hunk 5]
```diff
@@ -179,7 +179,7 @@ init([Call, Data]) ->
     DocId = ?MATCH_MODB_PREFIX(wh_util:to_binary(Year), wh_util:pad_month(Month), RecordingId),
     InteractionId = whapps_call:custom_channel_var(?CALL_INTERACTION_ID, Call),
     DefaultMediaName = get_media_name(wh_util:rand_hex_binary(16), Format),
-    MediaName = wh_json:get_value(?RECORDING_ID_KEY, Data, DefaultMediaName),
+    MediaName = wh_json:get_first_defined(?RECORDING_ID_KEY, Data, DefaultMediaName),
```
Comment (lazedo) [inline core/whistle_media/src/wh_media_recording.erl]: same here

File: doc/user_guides/call_recording.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+Key | Description | Type | Default | Required
+--- | ----------- | ---- | ------- | --------
+`action` | Toggle whether to start or stop a recording | `string('start', 'stop')` |   | `true`
+`format` | Disk Format to write the recording | `string('mp3', 'wav')` |   | `false`
+`record_min_sec` | Minimal record time, in seconds, to store recordings | `integer` |   | `false`
+`record_on_answer` | Whether to delay starting the recording until the call is answered | `boolean` |   | `false`
+`record_sample_rate` | Sampling rate of the recording, in Hz | `integer` |   | `false`
+`time_limit` | Limit, in seconds, of how long to record the call | `integer` |   | `false`
+`url` | What URL to use as a base for where to send the recording after it finishes | `string` |   | `false`
```
Comment (lazedo) [inline doc/user_guides/call_recording.md]: see previous comment on `url`

File: doc/user_guides/call_recording.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+##### Callflow action - Record Caller
+
+The `record_caller` callflow action is meant for recording things like voicemails or IVR prompts. The schema for the callflow action is:
+
+Key | Description | Type | Default | Required
+--- | ----------- | ---- | ------- | --------
+`format` | What format to store the recording on disk | `string('mp3', 'wav')` |   | `false`
+`method` | What HTTP method to use when sending the recording | `string('put', 'post')` |   | `false`
+`time_limit` | Time limit, in seconds, for the recording | `integer` | `3600` | `false`
+`url` | The URL to use when sending the recording for storage | `string` |   | `true`
```
Comment (lazedo) [inline doc/user_guides/call_recording.md]: same here

**Comment by onnet** [comment]
@jamesaimonetti Sorry in advance if it doesn't make sense, but while reading  doc/user_guides/call_recording.md it is not clear how and where to store settings like url, time_limit, record_on_answer, etc in case I would like to enable call recording by adding "record_call":true at Account, User, or Device level.

**Comment by jamesaimonetti** [comment]
@onnet Thanks for the feedback! I'll incorporate it when I revisit the comments here.

**Pull request merged**